File: simpleMediaPlayer/app/src/main/java/com/daeseong/simplemediaplayer/String_util.java
package com.daeseong.simplemediaplayer;

import android.text.TextUtils;

public class String_util {

    private String_util() {
    }

    //파일 확장자
    public static String getFileExt(String url){
        int index = url.lastIndexOf(".");
        return index >= 0 ? url.substring(index + 1) : "";
    }

    //파일 이름앞 폴더명
    public static String getLastFolderName(String url){
        if (TextUtils.isEmpty(url)) return "";

        int lastSlash = url.lastIndexOf("/");
        if (lastSlash >= 0) {
            int secondLastSlash = url.lastIndexOf("/", lastSlash - 1);
            return secondLastSlash >= 0 ? url.substring(secondLastSlash + 1, lastSlash) : "";
        }
        return "";
    }

    //파일 이름
    public static String getFileName(String url){
        int lastSlash = url.lastIndexOf("/");
        return lastSlash >= 0 ? url.substring(lastSlash + 1) : "";
    }
}

File: simpleMediaPlayer(kotlin)/app/src/main/java/com/daeseong/simplemediaplayer/String_util.kt
package com.daeseong.simplemediaplayer

object String_util  {

    //파일 확장자
    fun getFileExt(url: String): String {
        val index = url.lastIndexOf(".")
        return if (index >= 0) url.substring(index + 1) else ""
    }

    //파일 이름앞 폴더명
    fun getLastFolderName(url: String): String {
        if (url.isEmpty()) return ""

        val lastSlash = url.lastIndexOf("/")
        if (lastSlash >= 0) {
            val secondLastSlash = url.lastIndexOf("/", lastSlash - 1)
            return if (secondLastSlash >= 0) url.substring(secondLastSlash + 1, lastSlash) else ""
        }
        return ""
    }

    //파일 이름
    fun getFileName(url: String): String {
        val lastSlash = url.lastIndexOf("/")
        return if (lastSlash >= 0) url.substring(lastSlash + 1) else ""
    }
}